### PR TITLE
feat: As a team owner i can send or push tips to my team

### PR DESCRIPTION
**Pull request by farrellh1**
#### Tasks

- [ ] Build the frontend page : Send Tips
- [ ] Build the frontend page : Push Tips
- [x] Create Send / Push tips contract
- [ ] Integrate Web3 call to Send Tips and push Tips.

[Issue Link](https://github.com/globe-and-citizen/cnc-portal/issues/6)

### CODE REVIEW

**Comment by hermannleboss** [comment]
@farrellh1 your PR should respect a Single responsability principle : Make sure your pull request addresses a single issue


Here, your pull request is updating the frontend and in the same time it's creating the contract. 

It's not a good approach. 

For ex: if i think the smart contract is not good (or have a security issue), but i find the frontend good, how can i handle it ?

I should be able to validate a single feature or not.

File: contract/contracts/Tips.sol [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+        addressToTips[msg.sender] = 0;
+
+        emit WithdrawTip(msg.sender, tipEarned);
+    }
+
+    function getTips(address _address) external view returns (uint256) {
+      return addressToTips[_address];
+    }
+}
```
[the remark this answers is not in the document]
Comment (hermannleboss) [reply contract/contracts/Tips.sol:41]: Tips should be sent to a team: a team is a list of address.

File: contract/test/Tips.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+      await tips.pushTip(member1.address, { value: tipAmount });
+
+      const member1EndingBalance = await ethers.provider.getBalance(member1.address);
+      ;
+      expect(member1EndingBalance).to.equal(
+        member1StartingBalance + ethers.parseEther(tipAmount.toString())
+      );
+    });
+  });
```
[the remark this answers is not in the document]
Comment (hermannleboss) [reply contract/test/Tips.ts]: Your test is in typescript. But your project is just js. 
You don't select typescript when setting up the hardhat project

**Comment by farrellh1** [comment]
Cancelled, PR must one responsibillity

**Pull request closed without merging**